database/database_handler.py

```diff
@@ -37,31 +40,19 @@ def _get_db_access_token(self, client_id):
         return access_token
 
     def _get_database_connection_string(self):
-        """
-        Gets the connection string for the PostgreSQL database.
-
-        The method first checks if the code is running locally. If it is,
-        it retrieves the PostgreSQL credentials from environment variables.
-        Otherwise, it loads the credentials from a JSON file in the project root.
-        """
-        if os.environ.get('POSTGRES_USER'):
-            # Local run, use environment variables for credentials
-            db_url = f"postgresql://{os.environ['POSTGRES_USER']}:{os.environ['POSTGRES_PASSWORD']}@{os.environ['POSTGRES_HOST']}/{os.environ['POSTGRES_DB']}"
-        else:
-            # Production run, load credentials from the JSON file
-            with open('database.json') as f:
-                config = json.load(f)
-
-            # Retrieve values from the JSON
-            hostname = config["hostname"]
-            username = config["username"]
-            database_name = config["database_name"]
-            client_id = config["client_id"]
-
-            # Get the password using the client_id from a secure source
-            password = self._get_db_access_token(client_id)
-
-            db_url = f"postgresql://{username}:{password}@{hostname}/{database_name}"
+        # Get the connection string for the PostgreSQL database.
+
+        # Production run, load credentials from the JSON file
+        with open('database.json') as f:
+            config = json.load(f)
+
+        # Retrieve values from the JSON
+        client_id = config["client_id"]  # TODO get this from the Managed Identity name in code
+
+        # Get the password using the client_id from a secure source
+        password = self._get_db_access_token(client_id)
+
+        db_url = f"postgresql://{self.db_username}:{password}@{self.db_hostname}/{self.db_name}"
 
         return db_url
 
```

Comment:
I meant to do the parsing like:


```
@staticmethod
    def extract_upload_date(path):
        parts = path.split('/')
        date_time_str = parts[-2]

        try:
            image_upload_date = datetime.strptime(date_time_str, "%Y-%m-%d_%H:%M:%S")
            return image_upload_date
        except ValueError as e:
            LOGGER.info(f"Invalid folder structure, can not retrieve date: {date_time_str}")
            raise e
```
and use it like this in val.py:

`upload_date = DBConfigSQLAlchemy.extract_upload_date(image_path)
`